File: backend/src/main/java/com/qbonita/api/entity/Produto.java
package com.qbonita.api.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

import java.math.BigDecimal;

@Entity
@Table(name = "produtos")
public class Produto extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String nome;

    @Column(nullable = false, length = 2000)
    private String descricao;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal valor;

    @ManyToOne(optional = false, fetch = FetchType.EAGER)
    @JoinColumn(name = "categoria_id", nullable = false)
    private Categoria categoria;

    @Column(nullable = false, length = 2000)
    private String imagemPrincipal;

    @Column(length = 2000)
    private String imagemSecundaria;

    @Column(nullable = false)
    private boolean ativo = true;

    @Column(nullable = false)
    private boolean destaque = false;

    public Long getId() {
        return id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public BigDecimal getValor() {
        return valor;
    }

    public void setValor(BigDecimal valor) {
        this.valor = valor;
    }

    public Categoria getCategoria() {
        return categoria;
    }

    public void setCategoria(Categoria categoria) {
        this.categoria = categoria;
    }

    public String getImagemPrincipal() {
        return imagemPrincipal;
    }

    public void setImagemPrincipal(String imagemPrincipal) {
        this.imagemPrincipal = imagemPrincipal;
    }

    public String getImagemSecundaria() {
        return imagemSecundaria;
    }

    public void setImagemSecundaria(String imagemSecundaria) {
        this.imagemSecundaria = imagemSecundaria;
    }

    public boolean isAtivo() {
        return ativo;
    }

    public void setAtivo(boolean ativo) {
        this.ativo = ativo;
    }

    public boolean isDestaque() {
        return destaque;
    }

    public void setDestaque(boolean destaque) {
        this.destaque = destaque;
    }
}

File: frontend/lib/domain/produto.dart
/// Produto vindo da API (`/produtos`).
class Produto {
  const Produto({
    required this.id,
    required this.nome,
    required this.descricao,
    required this.valor,
    required this.categoriaId,
    required this.imagemPrincipal,
    this.imagemSecundaria,
    this.ativo = true,
    this.destaque = false,
    this.categoriaNome,
  });

  final int id;
  final String nome;
  final String descricao;
  final double valor;
  final int categoriaId;
  final String imagemPrincipal;
  final String? imagemSecundaria;
  final bool ativo;
  final bool destaque;

  /// Preenchido quando a API envia `categoria` embutida (exibição em cards).
  final String? categoriaNome;

  factory Produto.fromJson(Map<String, dynamic> json) {
    final cat = json['categoria'] as Map<String, dynamic>?;
    final cid = cat != null
        ? (cat['id'] as num).toInt()
        : (json['categoriaId'] as num).toInt();
    return Produto(
      id: (json['id'] as num).toInt(),
      nome: json['nome'] as String,
      descricao: json['descricao'] as String,
      valor: (json['valor'] as num).toDouble(),
      categoriaId: cid,
      imagemPrincipal: json['imagemPrincipal'] as String,
      imagemSecundaria: json['imagemSecundaria'] as String?,
      ativo: json['ativo'] as bool? ?? true,
      destaque: json['destaque'] as bool? ?? false,
      categoriaNome: cat?['nome'] as String?,
    );
  }
}
